src/Compilers/Core/Portable/Syntax/SyntaxList`1.cs

```diff
@@ -23,6 +24,42 @@ internal SyntaxList(SyntaxNode node)
             _node = node;
         }
 
+        /// <summary>
+        /// Creates a singleton list of syntax nodes.
+        /// </summary>
+        /// <param name="node">The single element node.</param>
+        public SyntaxList(TNode node)
+            : this((SyntaxNode)node)
+        {
+        }
+
+        /// <summary>
+        /// Creates a list of syntax nodes.
+        /// </summary>
+        /// <param name="nodes">A sequence of element nodes.</param>
+        public SyntaxList(IEnumerable<TNode> nodes)
+            : this(CreateNode(nodes))
+        {
+        }
+
+        private static SyntaxNode CreateNode(IEnumerable<TNode> nodes)
+        {
+            if (nodes == null)
+            {
+                return null;
```

Comment:
Why do we throw ArgumentNullException in SyntaxNodeOrTokenList.CreateNode, but return `null` here?